handout/testing/components/async.md

```diff
@@ -1,43 +1,78 @@
 # Testing Asynchronous Actions
 
-Sometimes we need to test components that rely on asynchronous actions that happen at specific times.  Angular provides a function called `fakeAsync` which wraps our tests in a zone and gives us access to the `tick` function, which will allow us to simulate the passage of time precisely.
+Sometimes we need to test components that rely on asynchronous actions that happen at specific times. Angular provides a function called `fakeAsync` which wraps our tests in a zone and gives us access to the `tick` function, which will allow us to simulate the passage of time precisely.
 
-Suppose we had a component with a button that will trigger an action within a setTimeout of two seconds. Here's what our component and unit test might look like:
+Let's go back to the example of the `QuoteComponent` component and rewrite the unit test using `fakeAsync`:
 
 ```js
+import { Component } from '@angular/core';
+import { QuoteService } from './quote.service';
+
 @Component({
-  selector: 'example',
-  template: `
-		<span>{{message}}</span>
-		<button (click)="performAction()">Click me</button>
-	`})
-
-class SampleComponent {
-  constructor() {}
-
-  performAction () {
-    setTimeout(() => {
-      this.message = 'My expected data';
-    }, 2000);
-  }
+  selector: 'my-quote',
+  template: '<h3>Random Quote</h3> <div>{{quote}}</div>'
+})
+
+export class QuoteComponent {
+  quote: string;
+
+  constructor(private quoteService: QuoteService){};
+
+  getQuote() {
+    this.quoteService.getQuote().then((quote) => {
+      this.quote = quote;
+    });
+  };
 }
 ```
 
 ```js
-it('Should work',
-  fakeAsync(inject([TestComponentBuilder], (tcb: TestComponentBuilder) => {
-    tcb.createAsync(SampleComponent).then(fixture => {
-      fixture.debugElement.nativeElement.querySelector('button')
-        .click();
+import { QuoteService } from './quote.service';
+import { QuoteComponent } from './quote.component';
+import { provide } from '@angular/core';
+import {
+  async,
+  TestBed,
+  fakeAsync,
+  tick,
+} from '@angular/core/testing';
 
-      tick(2000);
+class MockQuoteService {
+  public quote: string = 'Test quote';
 
-      fixture.detectChanges();
-      expect(fixture.debugElement.nativeElement.querySelector('span'))
-        .toHaveText('My expected data');
+  getQuote() {
+    return new Promise<string>((resolve, reject) => {
+      resolve(this.quote);
     });
```

Comment:
We could use the same simplification here of just using `return Promise.resolve(this.quote);`.